network/vif-route-qubes

```diff
@@ -173,29 +194,44 @@ fi
 
 # add anti-spoofing rules before enabling the interface
 if [ "${ip}" ]; then
+    peer_mac=$(xenstore-read -- "$XENBUS_PATH/mac") ||
+        fatal 'cannot obtain peer MAC from Xenstore'
+    readonly peer_mac
+    [[ "$peer_mac" =~ ^[0-9a-f]{2}(:[0-9a-f]{2}){5}$ ]] ||
+        fatal "bad mac address $peer_mac"
+    readonly iface_index_file="/var/run/xen/iface-$vif-index"
+    if [[ "$command" = 'online' ]]; then
+        read -r ifindex < "/sys/class/net/$vif/ifindex"
+        echo "$ifindex" > "$iface_index_file"
+    else
+        read -r ifindex < "$iface_index_file"
+        rm -f "$iface_index_file"
+    fi
+
     # If we’ve been given a list of IP addresses, then add routes from us to
     # the VMs we serve using those addresses.
     for addr in ${ip};
     do
         if [[ "$addr" = *:* ]]; then
-            ipt=ip6tables-restore
+            ipt=ip6
         else
-            ipt=iptables-restore
+            ipt=ip
         fi
-        printf '%s\n' "*raw" \
-            "$iptables_cmd -i ${vif} ! -s ${addr} -j DROP" \
-            "$iptables_cmd ! -i vif+ -s ${addr} -j DROP" \
-            "COMMIT" |
-            ${cmdprefix} "$ipt" --noflush $ipt_arg
         if ! conntrack_purge -s "$addr" || ! conntrack_purge -d "$addr"; then
             printf 'Cannot purge stale conntrack entries for %q\n' "$addr">&2
             exit 1
         fi
+        # Add neighbour entries
+        if [[ "$command" = 'online' ]]; then
+            ip -- neighbour add to "$addr" \
+                dev "$vif" lladdr "$peer_mac" nud permanent
+        fi
+        nft "$nftables_cmd element $ipt qbs-raw allowed { $ifindex . $addr }
+$nftables_cmd element $ipt qbs-raw downstream { $addr }"
     done
     # if no IPv6 is assigned, block all IPv6 traffic on that interface
     if ! [[ "$ip" = *:* ]]; then
-        echo -e "*raw\\n$iptables_cmd -i ${vif} -j DROP\\nCOMMIT" |
-            ${cmdprefix} ip6tables-restore --noflush $ipt_arg
+        echo 1 > "/proc/sys/net/conf/$vif/disable_ipv6"
```

Comment:
AFAIR there was a reason why it uses firewall, not disable_ipv6 flag. I don't remember details but it could be some applications failing when they prefer IPv6, or when using `::IPv4` style addresses.